### ISSUE TITLE
Completion detects non-flag argument as flag

### ISSUE THREAD

**Issue by oncilla** [spf13/cobra#1816]
We have encountered some odd behavior in the auto completion.
It appears that arguments with a dash as the second character are detected as flags by mistake.

The full reproducer with various test cases can be found here:
https://go.dev/play/p/UxqRkMcEX2a

The minimal reproducer is the following:

```go
package main

import "github.com/spf13/cobra"

func main() {
	root := &cobra.Command{
		Use: "root",
	}
	root.SetArgs([]string{"__complete", "1-ff00:0:1", ""})
	root.Execute()
}
```

which results in the following output:

```
[Debug] [Error] Subcommand 'root' does not support flag '1'
:0
Completion ended with directive: ShellCompDirectiveDefault
```

I tracked down the issue and appears that we enter this branch by mistake:
https://github.com/spf13/cobra/blob/fce8d8aeb08dc6afe413cc0af67a7fbb3cffec4c/completions.go#L577-L595

In particular, it appears that the is `isFlagArg` classifies the argument as a flag byistake:

https://github.com/spf13/cobra/blob/fce8d8aeb08dc6afe413cc0af67a7fbb3cffec4c/command.go#L629-L632

**Comment by marckhouzam** [comment]
Nice find @oncilla !  Up to posting a PR to fix this?

**Comment by oncilla** [comment]
@marckhouzam of course.

Fix is here: #1817
